packages/components/tests/lib/controllers/app-controller.test.js

```diff
@@ -1,839 +1,1389 @@
+  it('AppController - textSelectorParams: should return correct parameters no trigger is specified on mobile', async () => {
+    appC = AppController.jestCreate(uiState, { textQueryTriggerMobile: null })
+    await appC.init()
+    appC._platform = { isMobile: true, isDektop: false }
+    const [event, eventParams] = appC.textSelectorParams
+    expect(event.name).toBe('LongTap')
+    expect(eventParams).toBeUndefined()
+  })
 
-    expect(uiC.panel.panelData.notification.visible).toBeTruthy()
-    expect(uiC.panel.panelData.notification.important).toBeTruthy()
-    expect(uiC.panel.panelData.notification.showLanguageSwitcher).toBeFalsy()
-    expect(uiC.panel.panelData.notification.text).toEqual('fooImportant1')
+  it('AppController - textSelectorParams: should return correct parameters if some other trigger is specified on mobile', async () => {
+    const triggerParams = {
+      a: 'value of A',
+      b: 'value of B'
+    }
+    appC = AppController.jestCreate(uiState, { textQueryTriggerMobile: triggerParams })
+    await appC.init()
+    appC._platform = { isMobile: true, isDektop: false }
+    const [event, eventParams] = appC.textSelectorParams
+    expect(event.name).toBe('GenericEvt')
+    expect(eventParams).toBe(triggerParams)
+  })
 
-    expect(uiC.popup.vi.messages).toEqual([ 'fooImportant1' ])
+  it('AppController - textSelectorParams: should return correct parameters if mousemove is enabled', async () => {
+    appC = AppController.jestCreate(uiState, { textQueryTriggerMobile: null })
+    await appC.init()
+    appC._platform = { isMobile: false, isDektop: true }
+    Object.defineProperty(appC, 'isMousemoveEnabled', { get: () => true })
+    const [event, eventParams] = appC.textSelectorParams
+    expect(event.name).toBe('MouseMove')
+    expect(eventParams).toEqual(defaultMousemoveParams)
+  })
 
-    expect(uiC.popup.vi.popupData.notification.visible).toBeTruthy()
-    expect(uiC.popup.vi.popupData.notification.important).toBeTruthy()
-    expect(uiC.popup.vi.popupData.notification.showLanguageSwitcher).toBeFalsy()
-    expect(uiC.popup.vi.popupData.notification.text).toEqual('fooImportant1')
+  it('AppController - textSelectorParams: should return correct parameters for the "dblClick" trigger on desktop', async () => {
+    appC = AppController.jestCreate(uiState, { textQueryTriggerMobile: 'dblClick' })
+    await appC.init()
+    appC._platform = { isMobile: false, isDektop: true }
+    const [event, eventParams] = appC.textSelectorParams
+    expect(event.name).toBe('MouseDblClick')
+    expect(eventParams).toBeUndefined()
   })
 
-  it('6 AppController - showLanguageInfo methods', () => {
-    uiC.showLanguageInfo()
-    let languageName = AppController.getLanguageName(LMF.getLanguageIdFromCode(uiC.panel.options.items.preferredLanguage.currentValue)).name
+  it('AppController - textSelectorParams: should return correct parameters for the "dblclick" trigger on desktop', async () => {
```

Comment:
textQueryTriggerMobile should be textQueryTriggerDesktop?

Reply:
Will fix